### PR TITLE
To name of

### PR DESCRIPTION
**Pull request by asd-and-Rizzo**
#39

### CODE REVIEW

File: ExpressionToCodeTest/ExpressionWithNameTest.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			var actual = ExpressionWithName.ToNameOf(() => theVariable);
+			Assert.Equal("theVariable",actual);
+		}
+
+
+
+		[Fact]
+		public void TheMethod_ToNameOf()
+		{
+			Expression<Func<int, string, string>> theMethod = (x, y) => TheMethod(x, y);
```
Comment (EamonNerbonne) [inline ExpressionToCodeTest/ExpressionWithNameTest.cs]: Is it useful to support lambdas with arguments?  We could also permit the shorter `() => TheMethod`.

File: ExpressionToCodeTest/ExpressionWithNameTest.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			var actual = ExpressionWithName.ToNameOf(theMethod);
+			Assert.Equal("TheMethod",actual);
+		}
+
+		[Fact]
+		public void TheMethod_ToFullNameOf()
+		{
+			Expression<Func<int, string, string>> theMethod = (x, y) => ExpressionWithNameTest.TheMethod(x, y);
+			var actual = ExpressionWithName.ToFullNameOf(theMethod);
+			Assert.Equal("ExpressionWithNameTest.TheMethod(x, y)",actual);
```
Comment (EamonNerbonne) [inline ExpressionToCodeTest/ExpressionWithNameTest.cs]: Isn't this equivalent to `ToCode(theMethod.Body)`?

File: ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		/// <param name="expression"></param>
+		/// <returns></returns>
+		/// <example>
+		/// var example = "some text";
+		/// var name = toName( () => example);  // "example"
+		/// </example>
+		public static string ToNameOf<TResult>(this Expression<Func<TResult>> expression)
+		{
+			string value = ToNameOf(expression);
+			return value;
```
Comment (EamonNerbonne) [inline ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs]: inline return statement.

File: ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+		public static string ToNameOf<T1, T2, TResult>(this Expression<Func<T1, T2, TResult>> expression)
+		{
+			string value = ToNameOf(expression);
+			return value;
```
Comment (EamonNerbonne) [inline ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs]: inline return statement.

File: ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+		public static string ToNameOf<T1, TResult>(this Expression<Func<T1, TResult>> expression)
+		{
+			string value = ToNameOf(expression);
+			return value;
```
Comment (EamonNerbonne) [inline ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs]: inline return statement.

File: ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+		public static string ToNameOf<T>(this Expression<Action<T>> expression)
+		{
+			string value = ToNameOf(expression);
+			return value;
```
Comment (EamonNerbonne) [inline ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs]: inline return statement.

File: ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+
+		public static string ToNameOf(this Expression<Action> expression)
+		{
+			string value = ToNameOf(expression);
+			return value;
```
Comment (EamonNerbonne) [inline ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs]: inline return statement.

File: ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+			}
+			var methodCallExpression = expression.Body as MethodCallExpression;
+			if (methodCallExpression != null)
+			{
+				var methodName = methodCallExpression.Method.Name;
+				value = methodName;
+			}
+			if (value == null)
+				throw new ArgumentException("expression", "Unknown expression");
+			return value;
```
Comment (EamonNerbonne) [inline ExpressionToCodeLib/Unstable_v2_Api/ExpressionWithName.cs]: If you split this into three methods and combine their return values using `??` I believe the intent of this method would be clearer.

**Comment by asd-and-Rizzo** [comment]
I did what could. Regarding ToCode vs ToFullName (which is used in ToValued ).
I case it hand this is same. But there are differences. Need to think of ToFullName  behavior cause not consistent.

**Comment by EamonNerbonne** [comment]
Looks good!  I really need to find time to get the API in order to expose these new functions more consistently with the old stuff...

**Pull request merged**